### PR TITLE
HTBHF-1815 Return no_match for dwp and hmrc apis.

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

**Comment by mikeSmith08** [comment]
So much for having small PRs... :-(

**Comment by mikeSmith08** [comment]
Can we get all the usually checks on this repo before we approve and merge this PR please?

**Review by mikeSmith08 — CHANGES_REQUESTED** [review]
Can we get all the usually checks on this repo before we approve and merge this PR please?

**Comment by dwybourn** [comment]
> So much for having small PRs... :-(



> Considering almost all the code is boilerplate or copy/paste, it would have been more effort to do this in several PRs that just one PR.

**Comment by mikeSmith08** [comment]
> Considering almost all the code is boilerplate or copy/paste, it would have been more effort to do this in several PRs that just one PR.

You could easily done the first PR with all the boiler plate code with a complete no-op endpoint, if you remove all the DTOs and builders this PR would have been a lot smaller, would like to have looked at the DTOs and basic API without the boiler plate code around, that would not have been hard to split it into two PRs. There are 32 files in this PR, I only consider about 14 of those files to be boiler plate code, so the majority isn't actually boiler plate code.

**Comment by dwybourn** [comment]
> > Considering almost all the code is boilerplate or copy/paste, it would have been more effort to do this in several PRs that just one PR.
> 
> You could easily done the first PR wqith all the boiler plate code with a complete no-op endpoint, if you remove all the DTOs and builders this PR would have been a lot smaller, would like to have looked at the DTOs and basic API without the boiler plate code around, that would not have been hard to split it into two PRs.

Removed all the code accept the main method. This PR is now just the absolute minimum to have an application.

**Comment by mikeSmith08** [comment]
Perfect, thanks. Can we get the extra PR checks enable too (Snyk, LGTM etc), I'm not sure how to do it, maybe one for @YetAnotherMatt ?

**Comment by dwybourn** [comment]
> Perfect, thanks. Can we get the extra PR checks enable too (Snyk, LGTM etc), I'm not sure how to do it, maybe one for @YetAnotherMatt ?

Added LGTM, initial PR needs to be in for snyk as snyk needs to detect the framework.

File: .travis.yml
Comment (YetAnotherMatt) [inline .travis.yml]: we're not actually deploying here

**Review by YetAnotherMatt — APPROVED** [review]
Just to confirm, there's no controller yet?
We also need instructions for deploying in the readme in a later pR.

File: .travis.yml
Comment (dwybourn) [reply .travis.yml]: Updated to just `Build`

**Comment by dwybourn** [comment]
> Just to confirm, there's no controller yet?
> We also need instructions for deploying in the readme in a later pR.

Correct. Will add deploy instructions in next PR

**Review by mikeSmith08 — APPROVED** [review]

**Pull request merged**
